presto-main/src/main/java/com/facebook/presto/testing/LocalQueryRunner.java

```diff
@@ -685,7 +685,19 @@ public Lock getExclusiveLock()
             System.out.println(PlanPrinter.textLogicalPlan(plan.getRoot(), plan.getTypes(), metadata.getFunctionManager(), plan.getStatsAndCosts(), session, 0, false));
         }
 
-        SubPlan subplan = planFragmenter.createSubPlans(session, plan, true, WarningCollector.NOOP);
+        SubPlan subplan = planFragmenter.createSubPlans(
+                session,
+                plan,
+                true,
+                new PlanNodeIdAllocator()
```

Comment:
I think the usual pattern in Presto code base is to make PlanNodeIdAllocator an interface, and have SimplePlanNodeIdAllocator and MockPlanNodeIdAllocator ? :)

Reply:
Why we cannot just create a normal `PlanNodeIdAllocator` here ?

Reply:
> Why we cannot just create a normal PlanNodeIdAllocator here 

In theory you should always pass the same `PlanNodeIdAllocator` that was used in the logical planner. So the plan node ids don't clash.

But for the LocalQueryRunner specifically this is never supposed to be called (because for the LocalQueryRunner PlanFragmenter is pretty much no-op). 

So, instead of messing around many method signatures to properly pull down the `PlanNodeIdAllocator` i decided to pass something dummy, as it never gets called.

> and have SimplePlanNodeIdAllocator and MockPlanNodeIdAllocator

This implementation is more like `NotImplementedPlanNodeIdAllocator`. But since we need one only in one place i woudn't bother of creating a separate class for it.